packages/ListBox/src/components/Label/Label.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useListBox from "../../useListBox";
+import labelStyles from "./Label.styles";
+
+const propTypes = {
+  placeholder: PropTypes.string,
+};
+
+let lastKnownLabel = "";
+
+export default function Label(props) {
+  const { placeholder } = props;
+  const [state] = useListBox();
+  const { selectedOptions, isMulti, options } = state;
+
+  if (!isMulti && selectedOptions.length && state.options[selectedOptions[0]].preventDefaultOnSelect) {
+    if (!lastKnownLabel) return placeholder;
+
+    return lastKnownLabel;
+  }
+
+  function getListboxLabelForMulti() {
```

Comment:
Can this function take `selectedOptions` as an argument and be extracted from the `Label` component to make it more focussed on rendering? I find it disjointed to have an `if` then a `function` then another couple `ifs`.